### PR TITLE
Feature: Extend & refactor campaign code tool

### PR DESCRIPTION
**Pull request by rasmusguterstam**
## Included

- Refactor of campaigns tool (design & minor bugs)
- Support for creating different types of campaigns (not limited to Monthly Percentage)
- A more compact and concise table of existing campaigns

## To be added

This pull request is simply to add support for creating other campaigns. Next part is to add these campaigns and their corresponding forms.

### CODE REVIEW

File: src/features/tools/campaign-codes/CampaignCodeFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                ...filter,
+                activeTo,
+              })
+            }}
+          />
+        </div>
+        <Button
+          variation="primary"
+          onClick={() => {
+            setFilter(initialCampaignFilter)
```
Comment (AlexSJ1990) [inline src/features/tools/campaign-codes/CampaignCodeFilter.tsx:87]: Nice!

File: src/features/tools/campaign-codes/campaign-forms/monthly-percentage-form.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        placeholder="Months"
+        disabled={loading}
+        options={numberOfMonthsOptions}
+      />
+      <Spacing top={'small'} />
+      <Centered>
+        <Button
+          variation="primary"
+          loading={loading}
+          disabled={loading || !formLooksGood(formData)}
```
Comment (AlexSJ1990) [inline src/features/tools/campaign-codes/campaign-forms/monthly-percentage-form.tsx:129]: good idea 👍

File: src/features/tools/campaign-codes/components/ClearableDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        search
+        selection
+        selectOnBlur={false}
+        value={value}
+        options={options}
+        onChange={onChange}
+      />
+      {!disabled && value !== '' && onClear !== undefined && (
+        <Icon
+          name="x"
```
Comment (AlexSJ1990) [inline src/features/tools/campaign-codes/components/ClearableDropdown.tsx:45]: x?

Comment (rasmusguterstam) [reply src/features/tools/campaign-codes/components/ClearableDropdown.tsx:45]: Yup, a semantic-ui 'X' icon! Used as the "clear" button

**Pull request merged**